lib/internal/child_process.js

```diff
@@ -699,6 +702,19 @@ function setupChannel(target, channel, serializationMode) {
     });
   });
 
+  target.on('newListener', function() {
+    if (!target.channel) return;
+
+    const messages = target.channel[kPendingMessages];
+    if (!messages.size) return;
+
+    for (const messageParams of messages) {
+      process.nextTick(() => ReflectApply(target.emit, target, messageParams));
+    }
```

Comment:
Allocating a next tick on each iteration can be fairly expensive. Can the entire loop be executed in a single nextTick?

```suggestion
    process.nextTick(() => {
      for (const messageParams of messages) {
        ReflectApply(target.emit, target, messageParams);
      }
    });
```

Reply:
If we are doing this, we should probably move the `if(!message.size) return;` test inside the `nextTick` callback to avoid a race condition.

Reply:
fixed!